docs/software-architecture-and-design.md

```diff
@@ -98,44 +99,43 @@ The Field class represents the state of the physical field being played on, whic
 
 It is worth noting that constants are still useful, and should still be used whenever possible. If a value realistically doesn't need to be changed, it should be a constant (with a nice descriptive name) rather than a `Dynamic Parameter`. Having too many `Dynamic Parameters` is overwhelming because there are too many values to understand and change, and this can make it hard to tune values to get the desired behaviour while under pressure during a game.
 
 
+# Important Protobuf Messages
+These are [protobuf](https://developers.google.com/protocol-buffers/docs/cpptutorial) messages that we define and that are important for understanding how the [AI](#ai) works.
+
+## Primitives
+`TbotsProto::Primitive`s represent simple actions that a robot can do. It does not represent or include _how_ these things are done. Some examples are:
+* Moving in a straight line to a position
+* Pivoting around a point
+* Kicking the ball at a certain direction
+
+`Primitives` act as the abstraction between our [AI](#ai) and our robot firmware. It splits the responsibility such that the [AI](#ai) is responsible for sending a `Primitive` to a robot telling it what it wants it to do, and the robot is responsible for making sure it does what it's told. For every `Primitive` protobuf message, there is an equivalent `Primitive` implementation in our robot firmware. When robots receive a `Primitive` command, they perform their own logic and control in order to perform the task specified by the `Primitive`.
+
 ## Robot Status
-The `Robot Status` class contains information about the status of a single robot. Examples of the information they include are:
+The `TbotsProto::RobotStatus` protobuf message contains information about the status of a single robot. Examples of the information they include are:
 * Robot battery voltage
 * Whether or not the robot senses the ball in the breakbeam
 * The capacitor charge on the robot
 * The temperature of the dribbler motor
 
-Information received from the robots is stored in `Robot Status` objects so that the rest of the system can easily access and make sense of the information if necessary. For example, we monitor incoming `Robot Status` and display warnings in the [Visualizer](#visualizer) if anything looks wrong so we can be alerted. For example, during a game we may get a "Low battery warning" for a certain robot, and then we know to substitute it and replace the battery before it dies on the field.
-
+Information about the robot status is communicated and stored as `RobotStatus` protobuf messages. For example, we monitor incoming `RobotStatus` and display warnings in the [Visualizer](#visualizer) if anything looks wrong so we can be alerted. For example, during a game we may get a "Low battery warning" for a certain robot, and then we know to substitute it and replace the battery before it dies on the field.
```

Comment:
> For example, we monitor incoming `RobotStatus` and display warnings in the [Visualizer](#visualizer) if anything looks wrong so we can be alerted.

nit: this sentence can be a bit more clear

Reply:
fixed